test/CMakeLists.txt

```diff
@@ -54,6 +54,74 @@ add_executable(${test_target} ${SPARROW_TESTS_SOURCES})
 target_link_libraries(${test_target} PRIVATE sparrow doctest::doctest)
 add_test(NAME ${test_target} COMMAND ${test_target})
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    set(compiles_options
+        /permissive-
+        /WX # treat warnings as errors 
+        /W4 # Baseline reasonable warnings
+        # /w14242 # 'identifier': conversion from 'type1' to 'type1', possible loss of data
+        # /w14254 # 'operator': conversion from 'type1:field_bits' to 'type2:field_bits', possible loss of data
+        # /w14263 # 'function': member function does not override any base class virtual member function
+        # /w14265 # 'classname': class has virtual functions, but destructor is not virtual instances of this class may not be destructed correctly
+        # /w14287 # 'operator': unsigned/negative constant mismatch
+        # /w14296 # 'operator': expression is always 'boolean_value'
+        # /w14311 # 'variable': pointer truncation from 'type1' to 'type2'
+        # /w14545 # expression before comma evaluates to a function which is missing an argument list
+        # /w14546 # function call before comma missing argument list
+        # /w14547 # 'operator': operator before comma has no effect; expected operator with side-effect
+        # /w14549 # 'operator': operator before comma has no effect; did you intend 'operator'?
+        # /w14555 # expression has no effect; expected expression with side- effect
+        # /w14619 # pragma warning: there is no warning number 'number'
+        # /w14640 # Enable warning on thread un-safe static member initialization
+        # /w14826 # Conversion from 'type1' to 'type_2' is sign-extended. This may cause unexpected runtime behavior.
+        # /w14905 # wide string literal cast to 'LPSTR'
+        # /w14906 # string literal cast to 'LPWSTR'
+        # /w14928 # illegal copy-initialization; more than one user-defined conversion has been implicitly applied
+        # /we4289 # nonstandard extension used: 'variable': loop control variable declared in the for-loop is used outside the for-loop scope
+        # /we4244 # conversion from 'type1' to 'type_2', possible loss of data
+        /Wall # all warnings
```

Comment:
This one is *not* recommended (see https://github.com/cpp-best-practices/cppbestpractices/blob/master/02-Use_the_Tools_Available.md#msvc ).
I suspect removing this will make the rest work better.

Reply:
No because `/Wall` activates warning from dependencies, which is what the Turner points. It's not something we can easilly turn off without dependency-specific flags. That's why it's not recommended with msvc. It's not like gcc/clang's `-Wall` which does *not* activate all warnings, but only recomended ones.

Reply:
It is not enough to disable the dependencies included through `"` :/